tests/test_packages/test_skills/test_gym.py

```diff
@@ -115,51 +66,14 @@ def test_gym(self, pytestconfig):
         shutil.copyfile(file_src, file_dst)
 
         # change number of training steps
-        skill_config_path = Path(
-            self.t, self.agent_name, "vendor", "fetchai", "skills", "gym", "skill.yaml"
+        setting_path = "vendor.{}.skills.gym.handlers.gym.args.nb_steps".format(
+            FETCHAI_NAME
         )
-        skill_config = SkillConfig.from_json(yaml.safe_load(open(skill_config_path)))
-        skill_config.handlers.read("gym").args["nb_steps"] = 20
-        yaml.safe_dump(dict(skill_config.json), open(skill_config_path, "w"))
-
-        try:
-            process = subprocess.Popen(  # nosec
-                [
-                    sys.executable,
-                    "-m",
-                    "aea.cli",
-                    "run",
-                    "--connections",
-                    "fetchai/gym:0.1.0",
-                ],
-                stdout=subprocess.PIPE,
-                env=os.environ.copy(),
-            )
-            time.sleep(10.0)
-
-            # TODO: check the run ends properly
-
-        finally:
-            process.send_signal(signal.SIGINT)
-            process.wait(timeout=20)
-
-            if not process.returncode == 0:
-                poll = process.poll()
-                if poll is None:
-                    process.terminate()
-                    process.wait(2)
+        force_set_config(setting_path, 20)
 
-            os.chdir(self.t)
-            result = self.runner.invoke(
-                cli, [*CLI_LOG_OPTION, "delete", self.agent_name], standalone_mode=False
-            )
-            assert result.exit_code == 0
+        gym_aea_process = self.run_agent("--connections", "fetchai/gym:0.1.0")
+        time.sleep(10.0)
 
-    @classmethod
-    def teardown_class(cls):
-        """Teardowm the test."""
-        os.chdir(cls.cwd)
-        try:
-            shutil.rmtree(cls.t)
-        except (OSError, IOError):
-            pass
+        gym_aea_process.send_signal(signal.SIGINT)
+        gym_aea_process.wait(timeout=20)
```

Comment:
missing assertion on return code

Reply:
I know. That's why a todo was here, returncode != 0.

Reply:
Ok let's return to this in a separate PR